src/applications/yellow-ribbon/containers/SearchForm/index.jsx

```diff
@@ -83,10 +83,23 @@ export class SearchForm extends Component {
     this.setState({ [key]: 'unlimited' });
   };
 
-  onStateChange = key => event => {
+  onReactStateChange = key => event => {
     this.setState({ [key]: event.target.value });
   };
 
+  onCountryChange = event => {
+    const { value } = event.target;
+    const { country } = this.state;
+
+    // Escape early if value has not changed.
+    if (value === country) {
+      return;
```

Comment:
The change handler is called even when the value didn't actually change?

Reply:
Good question! Let me experiment real quick

Reply:
It is not! Great catch, updated!